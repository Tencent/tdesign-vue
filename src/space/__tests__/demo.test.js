/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import alignVue from '@/src/space/_example/align.vue';
import baseVue from '@/src/space/_example/base.vue';
import breakLineVue from '@/src/space/_example/break-line.vue';
import separatorVue from '@/src/space/_example/separator.vue';
import sizeVue from '@/src/space/_example/size.vue';
import verticalVue from '@/src/space/_example/vertical.vue';

const mapper = {
  alignVue,
  baseVue,
  breakLineVue,
  separatorVue,
  sizeVue,
  verticalVue,
};

describe('Space', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Space ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
