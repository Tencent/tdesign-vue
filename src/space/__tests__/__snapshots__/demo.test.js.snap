// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space Space alignVue demo works fine 1`] = `
<div
  class="t-space t-space-horizontal"
  style="gap: 16px; flex-wrap: wrap;"
>
  <div
    class="t-space-item"
  >
    <div
      class="space-border t-space t-space-align-start t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <div>
          start
        </div>
      </div>
      <div
        class="t-space-item"
      >
        <button
          class="t-button t-size-m t-button--variant-base t-button--theme-primary"
          type="button"
        >
          <span
            class="t-button__text"
          >
            Button
          </span>
        </button>
      </div>
      <div
        class="t-space-item"
      >
        <div
          class="space-block"
        />
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="space-border t-space t-space-align-center t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <div>
          center
        </div>
      </div>
      <div
        class="t-space-item"
      >
        <button
          class="t-button t-size-m t-button--variant-base t-button--theme-primary"
          type="button"
        >
          <span
            class="t-button__text"
          >
            Button
          </span>
        </button>
      </div>
      <div
        class="t-space-item"
      >
        <div
          class="space-block"
        />
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="space-border t-space t-space-align-end t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <div>
          end
        </div>
      </div>
      <div
        class="t-space-item"
      >
        <button
          class="t-button t-size-m t-button--variant-base t-button--theme-primary"
          type="button"
        >
          <span
            class="t-button__text"
          >
            Button
          </span>
        </button>
      </div>
      <div
        class="t-space-item"
      >
        <div
          class="space-block"
        />
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="space-border t-space t-space-align-baseline t-space-horizontal"
      style="gap: 16px;"
    >
      <div
        class="t-space-item"
      >
        <div>
          baseline
        </div>
      </div>
      <div
        class="t-space-item"
      >
        <button
          class="t-button t-size-m t-button--variant-base t-button--theme-primary"
          type="button"
        >
          <span
            class="t-button__text"
          >
            Button
          </span>
        </button>
      </div>
      <div
        class="t-space-item"
      >
        <div
          class="space-block"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Space Space baseVue demo works fine 1`] = `
<div
  class="t-space t-space-horizontal"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
</div>
`;

exports[`Space Space breakLineVue demo works fine 1`] = `
<div
  class="t-space t-space-horizontal"
  style="gap: 16px; flex-wrap: wrap;"
>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Button
      </span>
    </button>
  </div>
</div>
`;

exports[`Space Space separatorVue demo works fine 1`] = `
<div
  class="t-space t-space-align-center t-space-horizontal"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-text t-button--theme-default"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Text
      </span>
    </button>
  </div>
  <div
    class="t-space-item-separator"
  >
    <div
      class="t-divider t-divider--vertical"
    />
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-text t-button--theme-default"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Text
      </span>
    </button>
  </div>
  <div
    class="t-space-item-separator"
  >
    <div
      class="t-divider t-divider--vertical"
    />
  </div>
  <div
    class="t-space-item"
  >
    <button
      class="t-button t-size-m t-button--variant-text t-button--theme-default"
      type="button"
    >
      <span
        class="t-button__text"
      >
        Text
      </span>
    </button>
  </div>
</div>
`;

exports[`Space Space sizeVue demo works fine 1`] = `
<div>
  <div
    aria-valuetext="8"
    class="t-slider__container"
  >
    <div
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      class="t-slider"
      role="slider"
    >
      <div
        class="t-slider__rail"
      >
        <div
          class="t-slider__track"
          style="width: 0%; left: 0%;"
        />
        <div
          class="t-slider__button-wrapper"
          show-tooltip="true"
          style="left: 0%;"
          tabindex="0"
        >
          <div
            class="t-slider__button"
          />
        </div>
      </div>
    </div>
  </div>
   
  <br />
   
  <div
    class="t-space t-space-horizontal"
    style="gap: 8px;"
  >
    <div
      class="t-space-item"
    >
      <button
        class="t-button t-size-m t-button--variant-base t-button--theme-primary"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Button
        </span>
      </button>
    </div>
    <div
      class="t-space-item"
    >
      <button
        class="t-button t-size-m t-button--variant-base t-button--theme-primary"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Button
        </span>
      </button>
    </div>
    <div
      class="t-space-item"
    >
      <button
        class="t-button t-size-m t-button--variant-base t-button--theme-primary"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Button
        </span>
      </button>
    </div>
    <div
      class="t-space-item"
    >
      <button
        class="t-button t-size-m t-button--variant-base t-button--theme-primary"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Button
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Space Space verticalVue demo works fine 1`] = `
<div
  class="t-space t-space-vertical"
  style="gap: 16px;"
>
  <div
    class="t-space-item"
  >
    <div
      class="t-card t-card--bordered t-card--shadow-hover"
    >
      <div
        class="t-card__header"
      >
        <div
          class="t-card__header-wrapper"
        >
          <div>
            <span
              class="t-card__title"
            >
              标题
            </span>
          </div>
        </div>
        <div
          class="t-card__actions"
        >
          操作
        </div>
      </div>
      <div
        class="t-card__body"
      >
        
    仅有内容区域的卡片形式。卡片内容区域可以是文字、图片、表单、表格等形式信息内容。可使用大中小不同的卡片尺寸，按业务需求进行呈现。
  
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="t-card t-card--bordered t-card--shadow-hover"
    >
      <div
        class="t-card__header"
      >
        <div
          class="t-card__header-wrapper"
        >
          <div>
            <span
              class="t-card__title"
            >
              标题
            </span>
          </div>
        </div>
        <div
          class="t-card__actions"
        >
          操作
        </div>
      </div>
      <div
        class="t-card__body"
      >
        
    仅有内容区域的卡片形式。卡片内容区域可以是文字、图片、表单、表格等形式信息内容。可使用大中小不同的卡片尺寸，按业务需求进行呈现。
  
      </div>
    </div>
  </div>
  <div
    class="t-space-item"
  >
    <div
      class="t-card t-card--bordered t-card--shadow-hover"
    >
      <div
        class="t-card__header"
      >
        <div
          class="t-card__header-wrapper"
        >
          <div>
            <span
              class="t-card__title"
            >
              标题
            </span>
          </div>
        </div>
        <div
          class="t-card__actions"
        >
          操作
        </div>
      </div>
      <div
        class="t-card__body"
      >
        
    仅有内容区域的卡片形式。卡片内容区域可以是文字、图片、表单、表格等形式信息内容。可使用大中小不同的卡片尺寸，按业务需求进行呈现。
  
      </div>
    </div>
  </div>
</div>
`;
